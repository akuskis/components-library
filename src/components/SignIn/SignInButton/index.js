// @flow

import { Button } from '@material-ui/core';
import React from 'react';

import './signInButton.css';

type Props = {
  onClick: Function
};

const SignInButton = ({ onClick }: Props) => (
  <Button name="submit" className="sign-in-button" onClick={onClick}>
    ENTER
  </Button>
);

export default SignInButton;
