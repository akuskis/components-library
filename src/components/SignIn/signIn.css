.sign-in {
  display: flex;
  align-items: center;
  flex-direction: column;
}

.sign-in__logo {
  grid-area: logo;
  margin: 10px;
}

.sign-in__input {
  width: 450px;
  margin: 10px;
}

.sign-in__button {
  display: inline-flex;
  width: 175px;
  margin: 10px;
}
