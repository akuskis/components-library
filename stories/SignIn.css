.sign-in-story {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  display: flex;
  align-items: center;
  flex-direction: column;
}

.sign-in-story__form {
  width: 400px;
  margin-top: 20px;
}
