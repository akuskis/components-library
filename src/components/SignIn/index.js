// @flow

import { Input } from '@material-ui/core';
import React, { Component } from 'react';

import SignInButton from './SignInButton';
import './signIn.css';

type Props = {
  logo?: string,
  onSignIn: Function
};

type State = {
  login: string,
  password: string
};

class SignIn extends Component<Props, State> {
  static defaultProps = { logo: undefined };

  state = { login: '', password: '' };

  handleChange = (event: { target: { name: string, value: string } }) => {
    const {
      target: { name, value }
    } = event;

    this.setState({
      [name]: value
    });
  };

  handleSignIn = () => {
    const { login, password } = this.state;
    const { onSignIn } = this.props;

    onSignIn(login, password);
  };

  render() {
    const { logo } = this.props;
    const { login, password } = this.state;

    return (
      <div className="sign-in">
        {logo ? <img className="sign-in__logo" src={logo} alt="sign in" /> : null}
        <Input name="login" className="sign-in__input" placeholder="login" onChange={this.handleChange} value={login} />
        <Input
          name="password"
          className="sign-in__input"
          placeholder="password"
          type="password"
          onChange={this.handleChange}
          value={password}
        />
        <div className="sign-in__button">
          <SignInButton onClick={this.handleSignIn} />
        </div>
      </div>
    );
  }
}

export default SignIn;
