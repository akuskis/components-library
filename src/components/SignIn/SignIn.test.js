import React from 'react';
import { shallow } from 'enzyme';

import SignIn from '.';

describe('<SignIn />', () => {
  it('should render without props', () => {
    const component = shallow(<SignIn />);

    expect(component).toMatchSnapshot();
  });

  it('should render with all props', () => {
    const component = shallow(<SignIn onSignIn={jest.fn()} />);

    expect(component).toMatchSnapshot();
  });

  it('should call a function on sign in', () => {
    const credentials = {
      login: 'Wade',
      password: 'Wilson'
    };

    const doSignIn = jest.fn();
    const component = shallow(<SignIn onSignIn={doSignIn} />);

    component
      .find({ name: 'login' })
      .simulate('change', { target: { name: 'login', value: credentials.login } });
    component
      .find({ name: 'password' })
      .simulate('change', { target: { name: 'password', value: credentials.password } });
    component.find({ name: 'submit' }).simulate('click');

    expect(doSignIn).toHaveBeenCalledTimes(1);
    expect(doSignIn).toHaveBeenCalledWith({
      login: credentials.login,
      password: credentials.password
    });
  });
});
