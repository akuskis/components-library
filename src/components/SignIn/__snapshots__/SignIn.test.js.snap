// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> should render with all props 1`] = `
<div
  className="sign-in"
>
  <img
    alt="sign in"
    className="sign-in__logo"
    src="some.pic"
  />
  <WithStyles(Input)
    className="sign-in__input"
    name="login"
    onChange={[Function]}
    placeholder="login"
    value=""
  />
  <WithStyles(Input)
    className="sign-in__input"
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value=""
  />
  <div
    className="sign-in__button"
  >
    <SignInButton
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`<SignIn /> should render without props 1`] = `
<div
  className="sign-in"
>
  <WithStyles(Input)
    className="sign-in__input"
    name="login"
    onChange={[Function]}
    placeholder="login"
    value=""
  />
  <WithStyles(Input)
    className="sign-in__input"
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value=""
  />
  <div
    className="sign-in__button"
  >
    <SignInButton
      onClick={[Function]}
    />
  </div>
</div>
`;
